config/config.go

```diff
@@ -58,9 +59,15 @@ const (
 	errPipelineReceiverNotExists
 	errPipelineProcessorNotExists
 	errPipelineExporterNotExists
-	errUnmarshalError
 	errMissingReceivers
 	errMissingExporters
+	errUnmarshalErrorOnTopLevelSection
+	errUnmarshalErrorOnExtension
```

Comment:
Do we need these errors (line 65-70) to have different codes?

Reply:
I wanted to ensure in the tests that the error was coming from the correct component, ie.: I was afraid of reporting the same error for other wrong component and thus missing a case.